### PR TITLE
[v2] Auto generating docs and openapi specification using annotations

### PR DESCRIPTION
**Pull request by YoganshSharma**

### CODE REVIEW

File: .github/workflows/swagger_docs.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        run: |
+          cd ./docs || exit 1
+          git config user.email "admin@kossiitkgp.org"
+          git config user.name "koss-service"
+          git config gpg.format ssh
+          git config user.signingkey ~/.ssh/koss-service.pub
+
+          if [ -n "$(git status --porcelain swagger.yaml index.html)" ]; then
+            git add swagger.yaml index.html
+            git commit -S -m "Updated endpoints documentation"
```
Comment (harshkhandeparkar) [inline .github/workflows/swagger_docs.yml:71]: No git push?

**Comment by proffapt** [comment]
Trigger karke dekh

File: cmd/backend.go [hunk 2]
```diff
@@ -15,6 +15,13 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// main function
+//
+//	@title			Kwoc Backend
```
Comment (harshkhandeparkar) [inline cmd/backend.go:20]: KWoC.

File: cmd/backend.go [hunk 3]
```diff
@@ -15,6 +15,13 @@ import (
...
+//	@version		2.0.0
+//	@description	Kwoc Backend API
```
Comment (harshkhandeparkar) [inline cmd/backend.go:22]: KWoC.

File: cmd/backend.go [hunk 4]
```diff
@@ -15,6 +15,13 @@ import (
...
+//	@Host			kwoc-2022.fly.dev
```
Comment (harshkhandeparkar) [inline cmd/backend.go:23]: This link may change. Can this field be omitted?

File: controllers/student.go [hunk 5]
```diff
@@ -44,6 +44,21 @@ type StudentDashboard struct {
 	ProjectsWorked []ProjectDashboard `json:"projects_worked"`
 }
 
+// RegisterStudent godoc
+//
+//	@Summary		Register a student
+//	@Description	Register a new student with the provided details.
+//	@Accept			json
+//	@Produce		plain
+//	@Param			username	body		string	true	"The username of the student"
```
Comment (harshkhandeparkar) [inline controllers/student.go:53]: Is it possible to add the annotation to the struct and pass the struct here?

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: cmd/backend.go [hunk 6]
```diff
@@ -15,6 +15,12 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// main function
+//
+//	@title			KwoC Backend
```
Comment (harshkhandeparkar) [inline cmd/backend.go:20]: KWoC. :dead:

File: cmd/backend.go [hunk 7]
```diff
@@ -15,6 +15,12 @@ import (
...
+//	@version		2.0.0
+//	@description	KwoC Backend API
```
Comment (harshkhandeparkar) [inline cmd/backend.go:22]: Here, too.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

**Comment by YoganshSharma** [comment]
> Trigger karke dekh
can't find a way to manually trigger this
only other way is to push changes to `v2-dev`  or  `main`

**Comment by harshkhandeparkar** [comment]
> Trigger karke dekh

workflow agar default branch mein nahi hai to trigger kar sakte hain?

**Comment by YoganshSharma** [comment]
> workflow agar default branch mein nahi hai to trigger kar sakte hain?

 it should be possible to manually trigger as we are using `workflow_dispatch`  event. 
 but I can't find an option to run it

**Comment by harshkhandeparkar** [comment]
> 

or another test branch

**Comment by harshkhandeparkar** [comment]
@YoganshSharma you will make another PR right?

**Pull request closed without merging**